### ISSUE TITLE
Erlang 22.0.1 release left Bintray repository in an inconsistent state

### ISSUE THREAD

**Issue by TomaszKlosinski** [rabbitmq/erlang-debian-package#20]
## Update from 2021

An **update from 2021**: Bintray is [shutting down](https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/) (or already has).

There are two installation options available for these packages:

 * [Launchpad PPA](https://launchpad.net/~rabbitmq/+archive/ubuntu/rabbitmq-erlang)
 * [Cloudsmith.io](https://cloudsmith.io/~rabbitmq/repos/rabbitmq-erlang/setup/#formats-deb)

## Original Question

Hello, I've tried to follow this documentation:
https://www.rabbitmq.com/install-debian.html#apt-bintray

and I have a problem with installing erlang from this repo:
```
root@rabbitmq-test-01:~# cat /etc/apt/sources.list.d/dl_bintray_com_rabbitmq_debian.list
deb http://dl.bintray.com/rabbitmq/debian bionic main
deb http://dl.bintray.com/rabbitmq-erlang/debian bionic erlang-21.x
```
```
root@rabbitmq-test-01:~# apt update
Ign:1 http://dl.bintray.com/rabbitmq/debian bionic InRelease
Ign:2 http://dl.bintray.com/rabbitmq-erlang/debian bionic InRelease
Hit:3 http://archive.ubuntu.com/ubuntu bionic InRelease
Get:4 http://dl.bintray.com/rabbitmq/debian bionic Release [65.5 kB]
Get:5 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]
Err:6 http://dl.bintray.com/rabbitmq-erlang/debian bionic Release
  404  Not Found [IP: 35.157.207.108 80]
Get:7 http://archive.ubuntu.com/ubuntu bionic-backports InRelease [74.6 kB]
Get:8 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]
Reading package lists... Done
E: The repository 'http://dl.bintray.com/rabbitmq-erlang/debian bionic Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```

Actually if you check the link:
http://dl.bintray.com/rabbitmq-erlang/debian/dists/
There's no directory for `bionic`. 

Something seems to be broken...

**Comment by tarrenj** [comment]
@TomaszKlosinski Normally, you would look in https://dl.bintray.com/rabbitmq-erlang/ubuntu/dists for the `bionic` Release, however the `ubuntu` directory is gone now.

Also the only `debian` release supported is `buster`, which is planned for release around mid 2019...

[This](https://groups.google.com/d/msg/rabbitmq-users/Pabuzu0jajE/8m4gOHp4AwAJ) post in their user group leads me to believe that they were caught off guard by the early Erlang release.

Can any of the devs/team members comment on this with an ETA please?

**Comment by rockandska** [comment]
Hi there,

Already ping @michaelklishin yesterday on IRC / Slack and another user cross post on [ML](https://groups.google.com/d/msg/rabbitmq-users/Pabuzu0jajE/zbJn8WuNAgAJ) but didn't get any answer....

I can't do my install since yesterday....

**Comment by tarrenj** [comment]
Have tried irc, slack, and Twitter as well. Going to try the Google group
tomorrow.

**Comment by rockandska** [comment]
> Have tried irc, slack, and Twitter as well. Going to try the Google group tomorrow.

If the problem is related to Bintray again, i really hope that RabbitMq will move away from Bintray because I only have problems with it ( 502, [files who disappear](#1 ), etc.. )

**Comment by rguilmont** [comment]
I tried to contact them yesterday directly from Bintray... no reponse so far.

> Have tried irc, slack, and Twitter as well. Going to try the Google group tomorrow.

**Comment by TomaszKlosinski** [comment]
I'm not sure if it's only related to Bintray - I can't see the erlang deb packages at Package Cloud as well (there're only RPMs):
https://packagecloud.io/rabbitmq/erlang

To be frank, I can't find erlang-21.x for Ubuntu 18.04 anywhere at the moment. :-/

**Comment by rockandska** [comment]
> I'm not sure if it's only related to Bintray - I can't see the erlang deb packages at Package Cloud as well (there're only RPMs):
> https://packagecloud.io/rabbitmq/erlang
> 
> To be frank, I can't find erlang-21.x for Ubuntu 18.04 anywhere at the moment. :-/

Not sure at 100%, but debian erlang was never available on packagecloud and seems even not part of the [package pipeline](https://ci.rabbitmq.com/teams/main/pipelines/erlang-debian-package/)

**Comment by mccw** [comment]
I was able to install erlang 22 on two ubuntu servers yesterday before the files disappeared.

**Comment by kaikrueger** [comment]
The problem is exactly the same when using https://packagecloud.io/rabbitmq/erlang

```
Err:6 https://packagecloud.io/rabbitmq/erlang/ubuntu bionic Release
  404  Not Found [IP: 54.241.144.200 443]
Reading package lists... Done                     
E: The repository 'https://packagecloud.io/rabbitmq/erlang/ubuntu bionic Release' does not have a Release file.
```

Could some fix that problem?
Thanks

**Comment by tarrenj** [comment]
I want to highlight that the packages are still _technically_ available.  For example, [stretch](https://dl.bintray.com/rabbitmq-erlang/debian/pool/erlang/21.3.8.2-1/debian/stretch/), and [bionic](https://dl.bintray.com/rabbitmq-erlang/debian/pool/erlang/21.3.8.2-1/ubuntu/bionic/).

The missing Release.txt file is just a list that `apt` uses to map available packages to downloadable files.  We can still get the files and install them ourselves with `dpkg -i <packagefile>`.

I hope this helps some!

**Comment by tarrenj** [comment]
[This](https://www.rabbitmq.com/which-erlang.html) page suggests that we may be able to install Erlang from the [Erlang Solutions](https://packages.erlang-solutions.com/erlang/) package repositories.

I've always installed from the Rabbit Bintray, but if there are no changes to the package, then this should work fine.

Has anyone tested this?

Edit: From this repo's README.md:

> Erlang Solutions provides Debian packages for many Erlang/OTP minor releases. However, patch releases in each series are often delayed or never published. In other words, they follow the Erlang/OTP sources published on erlang.org. This repository strives to make the latest patch releases of Erlang/OTP easy to consume.

**Comment by michaelklishin** [comment]
Our team is at an off-site this week, so I am only seeing this now.

We will republish all packages but our team does not control when or how Bintray reindexes or rebuilds repositories. Those who are not happy with Bintray can use [Package Cloud](https://packagecloud.io/rabbitmq) to provision RabbitMQ and Erlang Solutions packages. We'd consider distributing our Erlang packages using GitHub and Package Cloud but I cannot make any promises. The Erlang team at Ericsson should get their patch release distribution game together.

**Comment by michaelklishin** [comment]
Our Erlang org has reached the storage quota on Bintray. We will remove some older versions to free up space.

This package is pro bono kind of project for our team so it's very hard to justify paying for a Bintray Enterprise plan and there are no "Open Source Plus" kind of plan on Bintray. A long term solution for these packages is TBD.

**Comment by michaelklishin** [comment]
We've freed space by deleting older patch versions. The plan is to introduce an older patch version cleanup job on Concourse. We already do this for alpha (development) builds but those are easy to remove as they lose their value rapidly. With GA releases there can always be users who still rely
on an older patch version, so a reasonable number of latest patches for each minor should be kept around. How many that is remains to be decided.

For the record, Bintray will only display a storage quota warning when you go above 85% or so, and only in the old UI. We are not aware of an API endpoint we could use for monitoring.

**Comment by michaelklishin** [comment]
We have contacted Bintray support since our org is rate limited. When it is unblocked we will republish 22.0.1 and 21.3.8.2 releases which will rebuild the repository. There is no ETA that we can provide but this issue will be updated.

**Comment by michaelklishin** [comment]
No response from Bintray support 5 hours later. Those affected can install Erlang from [other sources](https://www.rabbitmq.com/which-erlang.html#erlang-repositories).

**Comment by michaelklishin** [comment]
Our org has been unblocked. Now waiting for `22.0.1` and `21.3.8.2` releases to be rebuilt and republished so that the repo is reindexed and available again.

**Comment by michaelklishin** [comment]
There are two more 22.0.1 build jobs yet to run (Debian Buster and Stretch) but as far as I can tell the repository is available again. I will double check and close this issue if that's the case for 21.3.x.

**Comment by michaelklishin** [comment]
An **update from 2021**: Bintray is [shutting down](https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/) (or already has).

There are two installation options available for these packages:

 * [Launchpad PPA](https://launchpad.net/~rabbitmq/+archive/ubuntu/rabbitmq-erlang)
 * [Cloudsmith.io](https://cloudsmith.io/~rabbitmq/repos/rabbitmq-erlang/setup/#formats-deb)
